accepted/00074-retracted-updates-support.md

```diff
@@ -0,0 +1,326 @@
+  states now (`retracted`/`final`), this can be a boolean information telling
+  whether the patch is retracted or not.
+  We should implement this as a checkbox, allowing the user to change
+  the attribute (although the change should be disabled for vendor patches).
+  We need to take into account the possible consequences of modifying
+  a patch (maybe regenerating some data (`rhnServerNeededCache`,
+  regenerating repodata of all involved channels)).
+
+- Patch list (*Patches -> All/Relevant*): an icon in the list, if the
+  patch is retracted.
+
+- *System detail -> Software -> Packages -> List/Upgrade/Install*: a **fat**
+  warning icon in the list, if a package is contained in a retracted patch. The
+  warning should contain a popup text, telling that selecting that package
+  **installs** the package.
+
+- Package install/upgrade confirmation screens: if there are any packages
+  contained retracted patches, we should explicitly inform the user on these
+  screens too.
+
+- *System detail -> Software -> Patches*: an icon in the list, if the
+  patch is retracted. The warning should contain a popup text, telling that
+  selecting the patch for installation doesn't have any effect (see the details
+  about zypper above).
+
+- *Channel detail -> Patches*: an icon
+
+- *Channel detail -> Packages*: an icon
+
+Q: consider this alternative: toggle button `show retracted patches`/`show
+  packages from retracted` in the lists above.
+
+
+### XMLRPC
+The following endpoints must be updated:
+
+- `ErrataHandler.getDetails` the endpoint will expose the `status` field of an
+  Erratum
+
+- `ErrataHandler.setDetails` (if we allow modifying the flag) for non-vendor
+  channels the endpoint allows setting `status` to either `retracted` or `final`
+
+- Q: `ErrataHandler.create` shall we support creating retracted patches via API?
+  Likely not.
+
+- `PackageHandler.getDetails` add a boolean flag to the return value, that tells
+  if the package is part of a retracted patch
+
+- `SystemHandler.schedulePackageInstall` no changes needed. The endpoint takes
+  package ids, so the user most likely knows the details of the package (and
+  a possible presence of a retracted patch too)
+
+Q: todo verify that user can't get list of package ids without knowing the
+  retracted status somewhere
+
+- `SystemHandler.scheduleApplyErrata` no changes needed for the same
+  reason. Moreover, even if user schedules a retracted patch application, the
+  patch won't be installed (see the details about zypper above). We might inform
+  the user about it somehow.
+
+- Information about the presence of a retracted patch on a system/in a channel
+  can be retrieved via existing XMLRPC methods and `ErrataHandler.getDetails`.
+
+
+## Iteration 2 - Content Lifecycle management
+
+Enhance Content Lifecycle Manager and notifications.
+
+### Filters
+
+- add a filter that strips retracted patches from the sources
```

Comment:
Should it be a more generic filter, based on patch/package metadata? Maybe even add one more option to the "Patch (Keyword)" filter, where we have Reboot Required, Package Manager Restart Required, Other keyword <free text>? In fact, would this filter already work if I set "other keyword": "status=retracted" ?

Reply:
No please. We already have filters for other patch attributes and generalizing those is unrelated to this RFC. I don't know, how long would such a task take, so it should be done separately, if there is a need for it.